src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -28,14 +54,23 @@ public static LSP.Range TextSpanToRange(TextSpan textSpan, SourceText text)
             return LinePositionToRange(linePosSpan);
         }
 
-        public static LinePosition PositionToLinePosition(LSP.Position position)
+        public static Task<LSP.Location> DocumentSpanToLocationAsync(DocumentSpan documentSpan, CancellationToken cancellationToken)
         {
-            return new LinePosition(position.Line, position.Character);
+            return TextSpanToLocationAsync(documentSpan.Document, documentSpan.SourceSpan, cancellationToken);
         }
 
-        public static Task<LSP.Location> DocumentSpanToLocationAsync(DocumentSpan documentSpan, CancellationToken cancellationToken)
+        public static async Task<LSP.LocationWithText> DocumentSpanToLocationWithTextAsync(DocumentSpan documentSpan, ClassifiedTextElement text, CancellationToken cancellationToken)
         {
-            return TextSpanToLocationAsync(documentSpan.Document, documentSpan.SourceSpan, cancellationToken);
+            var sourceText = await documentSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+            var locationWithText = new LSP.LocationWithText
```

Comment:
return TextSpanToLocationWithText(DocumetSpan.SourceSpan, sourceText)?

Reply:
This is actually a different type than TextSpanToLocation.  I've renamed the other one to be a little more clear.